sycl/test-e2e/Matrix/Inputs/joint_matrix_bfloat16_impl.hpp

```diff
@@ -83,28 +94,37 @@ void test() {
   static constexpr size_t MATRIX_N = TN * 2;
   static constexpr size_t MATRIX_K = TK * 2;
   T A[MATRIX_M][MATRIX_K];
-  T B[MATRIX_K / 2][MATRIX_N * 2];
-  TResult C[MATRIX_M][MATRIX_N];
+  T B[MATRIX_K][MATRIX_N];
+
+  TAcc C[MATRIX_M][MATRIX_N];
   TResult D[MATRIX_M][MATRIX_N];
+  TResult DRef[MATRIX_M][MATRIX_N];
+
+  if (std::is_same_v<TAcc, bfloat16> || std::is_same_v<TResult, bfloat16>) {
+    matrix_rand<T>(MATRIX_M, MATRIX_K, (T *)A, T(1));
+    matrix_rand<T>(MATRIX_K, MATRIX_N, (T *)B, T(1));
+  } else {
+    matrix_fill(MATRIX_M, MATRIX_K, (T *)A,
+                [](int i, int j) { return (i == j) ? T(1) : T(0); });
+    matrix_fill(MATRIX_K, MATRIX_N, (T *)B, [](int i, int j) { return i; });
+  }
```

Comment:
Why is the different code needed for Acc and Res bfloat16? Why not always init to random values in -1..1 range?
Also, init with identity matrix (like in else branch for A) or with the same value across either row or column (like in the else branch for B) may hide bugs. It is better to use something more random (either rand, or filling with different values across both rows and column, using some f(i, j).